### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: 'hpo_terms.obo'

### ISSUE THREAD

**Issue by drewjbeh** [populationgenomics/talos#466]
Hi there,

When running the pipeline through Nextflow, it seems the hpo file can't be found during this command
`GeneratePanelData --input validation_phenopackets.json --hpo hpo_terms.obo --output validation_hpo_panel_data.json`
From the nextflow cofig, this parameter is set as such: `params.hpo = "hpo_terms.obo"`, however this file does not exist in the root directory (or anywhere in the repo for that matter). Is this supposed to be fetched from somewhere and my pipeline is not working properly, or is the reference to the file for Nextflow incorrect?

Thanks!

**Comment by drewjbeh** [comment]
Same applies for genes_to_phenotypes.txt and phenio.db. I found some download links for all files in `example_usage.sh` and while I appreciate that these should/can be sourced and specified depending on the project and data, perhaps for the test data run these can be located in the `nf_test_inputs/` folder.
It seems that there is an extra config.toml file that specified google storage locations to get these files but this does not seem to work within the Nextflow run

**Comment by MattWellie** [comment]
Hi @drewjbeh, apologies for the MASSIVE delay in getting back to this. We've had a flurry of feature requests and refinements prior to publication which have chewed up my time.

I've proposed a PR [here](https://github.com/populationgenomics/talos/pull/491) which would stash all these large files in git-lfs, so that they will be downloaded when this repository is cloned. The files in question are ~3GB in total, and one has to be unpacked to ~11GB, so committing them directly wasn't realistic. 

If/When I get some spare time I'm planning to evaluate an alternative library that requires far less input data, which should make deployment easier still.

This was definitely caused by a bit of an "it works on my machine" myopia when I was getting the nextflow implementation running. I'll try and do a better job of experimenting in a fresh repository before merging changes designed for exeryone 😓

**Comment by drewjbeh** [comment]
Hi @MattWellie, no worries, thanks for coming back. As I said, I solved this by getting some local copies of these inputs that, luckily, there were links to in other usage scripts. I guess the big one is the `phenio.db` right? Realised while I was making this issue that this would be a bottleneck due to size.
On the topic of inputs, I see there are a couple new inputs in newer commits like exomiser and splicevardb. Was wondering about these, where they contribute and to what parts of the category and filtering logic, and if simply using the full db as inputs for Talos would work. Maybe again these can be included in either the git-lfs or your alternative library idea?

**Comment by MattWellie** [comment]
Yeah phenio.db is a monster. The alternative to Semsimian + input files is [Hpo3](https://hpo3.readthedocs.io/en/stable/), which doesn't require any files; they're bundled into the library install. Unfortunately we really nailed our HPO matching thresholds based on all our validation datasets, and there isn't a 1:1 equivalence in functionality between the two libraries. If the deployment convenience is worth the headache of running all the validation again I might upgrade to that.

Aha! more un(der)documented features 😅 

For SpliceVarDB there's a conversion script [here](https://github.com/populationgenomics/talos/blob/main/src/talos/ConvertSpliceVarDb.py) to take the download directly from their site and parse it into a Hail Table for use in annotation.

for Exomiser it's a little more indirect. There's a script [here](https://github.com/populationgenomics/production-pipelines/blob/main/cpg_workflows/scripts/combine_exomiser_variant_tsvs.py) which I'm using to parse and aggregate the variant TSV outputs from Exomiser v14 into a Hail Table. I'll make an issue to port that script over to this repo and document both features.

**Comment by MattWellie** [comment]
Hi @drewjbeh , I've punched up the documentation a bit around these annotations, and now there's a more formal expectation of what to download and which folders to put it in. This includes the gnomAD data which is now publicly available on Zenodo 🎉 

I'm closing this as dealt with, though I appreciate that some stub files would make it easier to kick the tires on this process. Some of the data (like the massive phenio.db file) I wouldn't feel comfortable slicing into a more portable stub whilst maintaining a correct data structure :/
